cleo_plugins/FileSystemOperations/FileSystemOperations.cpp

```diff
@@ -114,6 +118,7 @@ class FileSystemOperations
     static OpcodeResult WINAPI opcode_0A9A(CRunningScript* thread)
     {
         OPCODE_READ_PARAM_FILEPATH(filename);
+        PATH_SECURITY_CHECK(filename);
```

Comment:
why not make it part of OPCODE_READ_PARAM_FILEPATH?

Reply:
As not all file path operations need check like that.
For example set work directory. It does not access that path in any way but will be later used with relative paths.

Reply:
we should not allow setting work directory outside of game directory or game settings either

Reply:
I do not know, but just act like that is not forbidden as it not accesses the location in any way. Later you might use relative paths like "Program Files\Gta\..."

Reply:
I don't see why a script would need to work with outside folders. But anyway we can restrict all filepaths by default and make exceptions on case by case basis for certain commands.

Reply:
Reworked